src/domdiv/inline_images.py

```diff
@@ -0,0 +1,272 @@
+        else:
+            raise Exception(f"DebtScale: Unsupported resolution: {resolution}")
+
+    def debt(self, debt, fontsize):
+        if debt and debt.isdigit():
+            debt_image = self.n_image.format(n=int(debt))
+        else:
+            debt_image = self.blank_image
+        image_path = str(resource_handling.get_image_filepath(debt_image))
+        width = self.width * fontsize
+        height = self.height * fontsize
+        return f'<img src="{image_path}" width="{width}" height="{height}" valign="{self.valign}" />'
+
+
+class SimpleInlineImage(InlineImage):
+    def __init__(self, width, height, valign, image: str):
+        super().__init__(width, height, valign)
+        self.image = image
+
+
+class PotionInlineImage(SimpleInlineImage):
+    def potion(self, fontsize):
+        image_path = str(resource_handling.get_image_filepath(self.image))
+        width = self.width * fontsize
+        height = self.height * fontsize
+        return f'<img src="{image_path}" width="{width}" height="{height}" valign="{self.valign}" />'
+
+
+class VpInlineImage(SimpleInlineImage):
+    def __init__(self, width, height, valign, image: str, fontsize_multiplier):
+        SimpleInlineImage.__init__(self, width, height, valign, image)
+        self.fontsize_multiplier = fontsize_multiplier
+
+    def vp(self, vp, fontsize):
+        image_path = str(resource_handling.get_image_filepath(self.image))
+        width = self.width * fontsize
+        height = self.height * fontsize
+        img = f'<img src="{image_path}" width="{width}" height="{height}" valign="{self.valign}" />'
+        if vp:
+            return nobr(
+                f"<b><font size={fontsize * self.fontsize_multiplier}>{vp}</font></b>{img}"
+            )
+        return img
+
+
+class SunInlineImage(SimpleInlineImage):
+    def sun(self, fontsize):
+        image_path = str(resource_handling.get_image_filepath(self.image))
+        width = self.width * fontsize
+        height = self.height * fontsize
+        return f'<img src="{image_path}" width="{width}" height="{height}" valign="{self.valign}" />'
+
+    def sunplus(self, fontsize):
+        image_path = str(resource_handling.get_image_filepath(self.image))
+        width = self.width * fontsize
+        height = self.height * fontsize
+        return nobr(
+            f'<b>+1</b><img src="{image_path}" width="{width}" height="{height}" valign="{self.valign}" />'
+        )
+
+
+class InlineImages(object):
+    def __init__(
+        self,
+        coin: dict[str, CoinInlineImage],
+        potion: dict[str, PotionInlineImage],
+        vp: dict[str, VpInlineImage],
+        debt: dict[str, DebtInlineImage],
+        sun: dict[str, SunInlineImage],
+    ):
+        self.coin = coin
+        self.potion = potion
+        self.vp = vp
+        self.debt = debt
+        self.sun = sun
+
+
+IMAGES = InlineImages(
+    coin={
+        "m": CoinInlineImage(1.2, 1, valign="-28%", resolution="low"),
```

Comment:
This valign value, together with the `autoLeading`, were the key pieces in getting the number on the coin image to line up with the surrounding text.

Reply:
The 1.2:1 aspect ratio is because the images include whitespace on the right hand side. I don't think we need that; it would be better for them to just be square.